components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -1877,10 +1877,8 @@ private void cloneOperationPoliciesToAPI(String oldAPIUuid, API newAPI,
                 for (OperationPolicy operationPolicy : operationPolicies) {
                     String clonedPolicyId;
                     if (!clonedPolicies.containsKey(operationPolicy.getPolicyId())) {
-                        OperationPolicyData apiSpecificOperationPolicy =
-                                apiMgtDAO.getAPISpecificOperationPolicyByPolicyID(operationPolicy.getPolicyId(),
-                                        oldAPIUuid, newAPI.getOrganization(), true);
-                        clonedPolicyId = apiMgtDAO.cloneOperationPolicy(newAPI.getUuid(), apiSpecificOperationPolicy);
+                        clonedPolicyId = operationPolicyProviderInstance.cloneAPISpecificOperationPolicy(operationPolicy.getPolicyId(), oldAPIUuid,
```

Comment:
Isn't this a bit confusing? We are calling `cloneAPISpecificOperationPolicy()`, but what it returns is an ID. Where does the cloned policy go?

Reply:
when cloning, It makes a copy (a cloned) of the policy from one API to another API  and returns the ID of that newly saved policy.